ccvm_simulators/solvers/dl_solver.py

```diff
@@ -228,6 +276,83 @@ def tune(self, instances, post_processor=None, pump_rate_flag=True, g=0.05):
         #       future consideration
         self.is_tuned = True
 
+    def optics_machine_energy(self, machine_parameters=None):
+        """The wrapper function of calculating the maximum energy consumption of the
+        solver simulating on a DL-CCVM machine.
+
+        Args:
+            machine_parameters (dict, optional): Parameters of the. Defaults to None.
+
+        Raises:
+            ValueError: when the given machine parameters are not valid.
+            ValueError: when the given dataframe does not contain the required columns.
+
+        Returns:
+            Callable: A callable function that takes in a dataframe and problem size and
+                returns the maximum energy consumption of the solver.
+        """
+
+        if machine_parameters is None:
+            machine_parameters = self._default_optics_machine_parameters
+        else:
+            self._is_valid_optics_machine_parameters(machine_parameters)
```

Comment:
I didn't see an equivalent of this function for CPU or CUDA. Should we have one in CCVMSolver?

Reply:
Hmm. They are located in the CCVMSolver class. Please let me know if you still couldn't find them.